src/riscv/compiler.rs

```diff
@@ -695,7 +695,7 @@ fn process_instruction(instr: &str, args: &[Argument]) -> String {
         }
         "not" => {
             let (rd, rs) = rr(args);
-            format!("{rd} <=X= xor({rs}, 0xffffffff);\n")
+            format!("{rd} <=X= wrap_signed(-{rs} - 1);\n")
```

Comment:
Can `rs` overflow if it's negative?

Reply:
registers should always contain unsigned 32 bit values. If the value is too small or too large, I think `wrap_signed` will fail

Reply:
of course free inputs can have values outside the range.